src/plugins/intel_cpu/src/nodes/matmul.cpp

```diff
@@ -342,12 +314,18 @@ void MatMul::getSupportedDescriptors() {
         outPortPrec = firstInPortPrec = secondInPortPrec = Precision::FP32;
     }
 
+    Precision postOpsPrec = outPortPrec;
     if (!fusedWith.empty()) {
-        outPortPrec = fusedWith[fusedWith.size() - 1]->getOriginalOutputPrecisionAtPort(0);
+        postOpsPrec = fusedWith[fusedWith.size() - 1]->getOriginalOutputPrecisionAtPort(0);
     }
 
-    if (!canBeExecutedInInt8(firstInPortPrec, secondInPortPrec) && one_of(outPortPrec, Precision::U8, Precision::I8))
-        outPortPrec = Precision::FP32; // INT output is not supported for non-INT inputs
+    if (canBeExecutedInInt8(firstInPortPrec, secondInPortPrec)) {
+        // INT8 mode support wide range of output precisions
+        outPortPrec = postOpsPrec;
+    } else {
+        // non-INT8 mode only support limited precision
+        // we ignore postOpsPrec and stay with getOriginalOutputPrecisionAtPort(0)
```

Comment:
What if the `postOpsPrec` is `fp32` and `outPortPrec` is `bf16`? In that case we will have unnecessary rounding, which may affect the accuracy.

Reply:
your concern is valid, I was assuming `postOpsPrec` to be `int8` when wrote this logic, will change